File: 04-Kuku/Answers/Python/kuku3-ans.py
#!usr/bin/python
# -*- coding: UTF-8 -*-

# kuku2-ans.py
# By kota ninomiya

from math import floor, log10

def create_kuku_table_auto_format(number:int):
    """九九の表を作成する関数
    
    Args:
        number (int): 上限となる数字
    """
    message = ''
    term_digit = floor(log10(number-1))
    answer_digit = floor(log10((number-1)**2))
    for i in range(1, number): 
        for j in range(1, number):
            message += '{}{}*'.format(auto_format_digits(i, term_digit), i)
            message += '{}{}='.format(auto_format_digits(j, term_digit), j)
            message += '{}{}, '.format(auto_format_digits(i*j, answer_digit), i*j)
        message = message.rstrip(', ')
        message += '\n'
    print(message)


def auto_format_digits(verify_value:int, digit:int):
    """桁数をスペースで埋め合わせる関数
    
    Args:
        verify_value (int): 検証する値
        digit (int): 桁数
    """
    numOfFill = digit - floor(log10(verify_value))
    spaces  = ''
    for i in range(numOfFill):
        spaces += ' '
    return spaces


def main():
    create_kuku_table_auto_format(11)
        


if __name__ == '__main__':
    main()

File: 04-Kuku/Answers/JavaScript/kuku3-ans.js
// kuku3-ans.js
// By kota ninomiya

/**
 * 九九の表を作る関数，桁数自動調整版
 * @param {Number} number 上限となる数字
 */
function createKukuTableAutoFormat(number){
    message = ''
    termDigit = Math.floor(Math.log10((number-1)))
    answerDigit = Math.floor(Math.log10((number-1)*(number-1)))
    for(let i=1; i<number; i++){
        for(let j=1; j<number; j++){
            message += `${autoFormatDigits(i, termDigit) + i}*${autoFormatDigits(j, termDigit) + j}=${autoFormatDigits(i*j, answerDigit)}${i*j}, `
        }
        message = message.slice(0, -2)
        message += `\n`
    }
    console.log(message)
}

/**
 * 桁数にあわせてスペースで埋め合わせる関数
 * @param {Number} verifyValue 検証する値
 * @param {Number} digit 揃える桁数
 */
function autoFormatDigits(verifyValue, digit){
    numOfFill = digit - Math.floor(Math.log10(verifyValue))
    spaces = ''
    for(let i=0; i<numOfFill; i++){
        spaces += ' '
    }
    return spaces
}

createKukuTableAutoFormat(11)